swfo/h5utils.py

```diff
@@ -2,30 +2,47 @@
 Utility functions used in the conversion process to hdf5 archives
 """
 
-from typing import Optional, List, Dict, Union
-from io import BufferedReader
+from typing import Optional, List, Dict
+from io import StringIO, BufferedReader
+import os
 import uuid
 import urllib.parse
 import hashlib
-
+from pathlib import Path
+import tempfile
+from contextlib import contextmanager
 
 from ruamel.yaml import YAML as _YAML
-from ruamel.yaml.compat import StringIO
 import h5py
 
-VLEN_STRING = h5py.special_dtype(vlen=str)
+
 FALLBACK_UUID_NAMESPACE = uuid.UUID('c5908e58-7301-4054-9f04-a0fa8cdef63b')
+PUBLIC_NAMESPACE = 'METADATA'
+PRIVATE_NAMESPACE = '.METADATA'
+METADATA_PTR = 'CURRENT'
```

Comment:
doesn't (or maybe shouldn't) `wagl.constants` have these constants defined?

Reply:
Private/Public namespacing was introduced to give us the flexibility to rewrite the metadata docs without hard-locking the auxiliary directories.
There may be changes to the logic to calculate the spatial extents, update the format, increase the number of parameters/properties, or just changes towards adopting the dataset metadata tooling under development by Jeremy.

I'm not sure if we wish to extend this methodology towards our (currently) internal representation of the surface reflectance product prior to geotiff and yaml packaging (the h5 collection). 
As such it feels simpler to define these variables in SWFO until maturity is reached with handling the collection upgrade and the desired format for metadata serialisation and fields from a datacube perspective.